### PR TITLE
Redux application

### PR DESCRIPTION
**Pull request by Joanartdacosta**
applying redux in MenuDetailsId and tripsId:
- redux toolkit installation
- new reducer (tripSlice and saveTrip action)
- add store
- delete most of the props

### CODE REVIEW

File: app/components/common/menu_id/classification/Classification.js [hunk 1]
```diff
@@ -1,113 +1,123 @@
...
+    current = useSelector((state) => state.trip.selectedTrip);
+  } else if (props.page === "tour") {
+    current = useSelector((state) => state.tour.selectedTour);
+  }
+
   return (
     <div className="p-8 m-auto">
       <div>
-        {props.percentageAcessibility ? (
+        {current.percentageAcessibility ? (
```
Comment (doublethink13) [inline app/components/common/menu_id/classification/Classification.js:15]: what if current is null?

`current?.percentageAccessibility ?`

Comment (doublethink13) [reply app/components/common/menu_id/classification/Classification.js:15]: please change this everywhere else

Comment (Joanartdacosta) [reply app/components/common/menu_id/classification/Classification.js]: well noted! Thank you
Changes done

File: app/components/common/menu_id/general_info/Introduction.js [hunk 2]
```diff
@@ -1,14 +1,23 @@
...
       <div className="m-auto text-gray-900  dark:text-white">
         <div className="flex flex-col items-center justify-center">
-          <div className="mb-8 mt-8 text-3xl font-extrabold">{props.title}</div>
+          <div className="mb-8 mt-8 text-3xl font-extrabold">
+            {current.title}
+          </div>
 
           <div className="mb-8 text-gray-500 dark:text-gray-400">
-            {props.introduction}
+            {current.introduction}
```
Comment (doublethink13) [inline app/components/common/menu_id/general_info/Introduction.js:20]: current can be null

File: app/components/common/menu_id/location/Map.js [hunk 3]
```diff
@@ -25,8 +33,8 @@ export default function Map(props) {
 
         new GOOGLE.maps.Marker({
           position: {
-            lat: props.latitude,
-            lng: props.longitude,
+            lat: current.latitude,
```
Comment (doublethink13) [inline app/components/common/menu_id/location/Map.js:36]: current can be null

File: app/components/common/menu_id/location/Map.js [hunk 4]
```diff
@@ -25,8 +33,8 @@ export default function Map(props) {
...
+            lng: current.longitude,
```
Comment (doublethink13) [inline app/components/common/menu_id/location/Map.js:37]: current can be null

File: app/components/common/menu_id/path/PathOptions.js [hunk 5]
```diff
@@ -17,42 +25,43 @@ export default function PathOptions(props) {
       >
         <div className="py-5 border-b border-gray-200 dark:border-gray-700">
           <p className="mb-2 text-gray-500 dark:text-gray-400">
-            {props.description1}
+            {current.description1}
```
Comment (doublethink13) [inline app/components/common/menu_id/path/PathOptions.js:28]: current can be null

File: app/components/common/menu_id/path/PathOptions.js [hunk 6]
```diff
@@ -17,42 +25,43 @@ export default function PathOptions(props) {
...
           </p>
           <p className="mb-2 text-gray-500 dark:text-gray-400">
-            {props.waydescription}
+            {current.waydescription}
```
Comment (doublethink13) [inline app/components/common/menu_id/path/PathOptions.js:31]: current can be null

File: app/components/common/menu_id/path/PathOptions.js [hunk 7]
```diff
@@ -61,37 +70,42 @@ export default function PathOptions(props) {
         </div>
 
         <Path
```
Comment (doublethink13) [inline app/components/common/menu_id/path/PathOptions.js:72]: current can be null

File: app/components/common/menu_id/path/PathOptions.js [hunk 8]
```diff
@@ -61,37 +70,42 @@ export default function PathOptions(props) {
...
         ) : (
           ""
         )}
-        {props.accessOption3 ||
-        props.labelOption3Description ||
-        props.imagePath3 ? (
+        {current?.accessOption3 ||
+        current?.labelOption3Description ||
+        current?.imagePath3 ? (
           <Path
```
Comment (doublethink13) [inline app/components/common/menu_id/path/PathOptions.js:103]: current can be null

File: app/components/common/menu_id/general_info/Introduction.js [hunk 9]
```diff
@@ -1,14 +1,23 @@
...
+    current = useSelector((state) => state.tour.selectedTour);
+  }
+
   return (
     <div className="">
       <div className="m-auto text-gray-900  dark:text-white">
         <div className="flex flex-col items-center justify-center">
-          <div className="mb-8 mt-8 text-3xl font-extrabold">{props.title}</div>
+          <div className="mb-8 mt-8 text-3xl font-extrabold">
+            {current.title}
```
Comment (doublethink13) [inline app/components/common/menu_id/general_info/Introduction.js:16]: current can be null

File: app/components/common/menu_id/general_info/Tips.js [hunk 10]
```diff
@@ -1,15 +1,22 @@
...
+  let current = null;
+  if (props.page === "trip") {
+    current = useSelector((state) => state.trip.selectedTrip);
+  } else if (props.page === "tour") {
+    current = useSelector((state) => state.tour.selectedTour);
+  }
+
   return (
     <div>
       <InfoBullets
```
Comment (doublethink13) [inline app/components/common/menu_id/general_info/Tips.js:14]: current can be null

File: app/components/common/menu_id/path/PathOptions.js [hunk 11]
```diff
@@ -61,37 +70,42 @@ export default function PathOptions(props) {
...
+          imagePath1={current.imagePath1}
+          imagePath11={current.imagePath11}
+          page={props.page}
         />
 
-        {props.accessOption2 || props.labelOption2Description || imagePath2 ? (
+        {current?.accessOption2 ||
+        current?.labelOption2Description ||
+        imagePath2 ? (
           <Path
```
Comment (doublethink13) [inline app/components/common/menu_id/path/PathOptions.js:85]: current can be null

File: app/components/common/menu_id/general_info/Introduction.js (hunk 2 above)
Comment (Joanartdacosta) [reply app/components/common/menu_id/general_info/Introduction.js]: done

File: app/components/common/menu_id/location/Map.js (hunk 4 above)
Comment (Joanartdacosta) [reply app/components/common/menu_id/location/Map.js:37]: done

File: app/components/common/menu_id/path/PathOptions.js (hunk 7 above)
Comment (Joanartdacosta) [reply app/components/common/menu_id/path/PathOptions.js:72]: done

File: app/components/common/menu_id/general_info/Introduction.js (hunk 9 above)
Comment (Joanartdacosta) [reply app/components/common/menu_id/general_info/Introduction.js]: done

File: app/components/common/menu_id/general_info/Tips.js (hunk 10 above)
Comment (Joanartdacosta) [reply app/components/common/menu_id/general_info/Tips.js]: done

File: app/components/common/menu_id/path/PathOptions.js (hunk 11 above)
Comment (Joanartdacosta) [reply app/components/common/menu_id/path/PathOptions.js:85]: done

File: app/components/common/menu_id/location/Map.js (hunk 3 above)
Comment (Joanartdacosta) [reply app/components/common/menu_id/location/Map.js:36]: done

File: app/components/common/menu_id/path/PathOptions.js (hunk 5 above)
Comment (Joanartdacosta) [reply app/components/common/menu_id/path/PathOptions.js]: done

**Pull request merged**